incidents/incident_script.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Source code based off of
+# https://github.com/googleinterns/cloud-monitoring-notification-delivery-integration-sample-code/blob/server-setup/main.py
+
+"""Use custom metric to trigger and resolve incidents.
+
+This program can perform 3 operations depending on
+the argument that is given: create a new custom metric
+called "testing_metric" in the current GCP project,
+trigger an incident, or resolve the incident. To use
+it correctly, first create the custom metric (only
+do this once). Then in the GCP website create a policy
+where the metric is "testing_metric" and the resource
+type is "GCE VM Instance", and set it to trigger when the
+most recent value of any time series is above 3.0 (only create
+the policy once). Once the custom metric and policy are
+created, the program can properly trigger and resolve an
+incident on command. Note, that if you trigger an incident,
+you must resolve the incident before you can trigger it again.
+
+
+  How to use:
+
+  $ python3 incident_script.py -h
+  $ python3 incident_script.py create-custom-metric
+  $ python3 incident_script.py trigger-incident
+  $ python3 incident_script.py resolve-incident
+"""
+
+import argparse
+import os
+import time
+
+from google.cloud import monitoring_v3
+
+
+PROJECT_ID = os.environ['GOOGLE_CLOUD_PROJECT']
+METRIC_NAME = 'testing_metric'
```

Comment:
Good to know! And just to clarify are you saying that both PROJECT_ID and METRIC_NAME should be taken in as command line arguments or just METRIC_NAME? I'm just asking since the PROJECT_ID as of now is essentially receiving its value based on which project is already open in the command line.

Reply:
Just the metric name. Project ID is fine, since that'll change if you just change which you have open.

Reply:
Thanks, I just changed it.